pkg/virt-launcher/virtwrap/network/network_test.go

```diff
@@ -20,68 +20,62 @@
 package network
 
 import (
-	"os"
-
-	"github.com/golang/mock/gomock"
 	. "github.com/onsi/ginkgo"
 	. "github.com/onsi/gomega"
 
-	"kubevirt.io/kubevirt/pkg/virt-launcher/virtwrap/network/cache"
-
 	"kubevirt.io/kubevirt/pkg/virt-launcher/virtwrap/network/cache/fake"
 
 	v1 "kubevirt.io/client-go/api/v1"
 )
 
-var _ = Describe("Network", func() {
-	var mockpodNIC *MockpodNIC
-	var ctrl *gomock.Controller
-	var pid int
-	var cacheFactory cache.InterfaceCacheFactory
-
-	BeforeEach(func() {
-		pid = os.Getpid()
-		ctrl = gomock.NewController(GinkgoT())
-		mockpodNIC = NewMockpodNIC(ctrl)
-		cacheFactory = fake.NewFakeInMemoryNetworkCacheFactory()
-		podNICFactory = func(handler NetworkHandler, cacheFactory cache.InterfaceCacheFactory) podNIC {
-			return mockpodNIC
-		}
-	})
-	AfterEach(func() {
-		podNICFactory = newpodNIC
-	})
-
+var _ = Describe("VMNetworkConfigurator", func() {
 	Context("interface configuration", func() {
 		It("should configure bridged pod networking by default", func() {
 			vm := newVMIBridgeInterface("testnamespace", "testVmName")
+
+			vmNetworkConfigurator := NewVMNetworkConfigurator(vm, fake.NewFakeInMemoryNetworkCacheFactory())
 			iface := v1.DefaultBridgeNetworkInterface()
 			defaultNet := v1.DefaultPodNetwork()
-
-			mockpodNIC.EXPECT().PlugPhase1(vm, iface, defaultNet, primaryPodInterfaceName, pid)
-			err := SetupPodNetworkPhase1(vm, pid, cacheFactory)
-			Expect(err).To(BeNil())
+			nics, err := vmNetworkConfigurator.getNICs()
+			Expect(err).ToNot(HaveOccurred())
+			Expect(nics).To(ConsistOf([]podNIC{{
```

Comment:
From our discussion I get this is used to ignore the order in slices. Here you don't need it, since your array features a single item.

Only consider changing if you need to update more things.